File: viewjava/app/src/main/java/pers/shawxingkwok/kdatastore/viewjava/app/MainFragment.java
package pers.shawxingkwok.kdatastore.viewjava.app;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.os.Bundle;
import android.widget.Toast;
import androidx.annotation.IdRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import pers.shawxingkwok.kdatastore.viewjava.app.databinding.FragmentMainBinding;
import pers.shawxingkwok.kdatastore.viewjava.settings.Settings;
import pers.shawxingkwok.kdatastore.viewjava.settings.Theme;

public class MainFragment extends Fragment {

    private FragmentMainBinding binding;

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Nullable
    @Override
    public View onCreateView(
        @NonNull LayoutInflater inflater,
        @Nullable ViewGroup container,
        @Nullable Bundle savedInstanceState
    ) {
        binding = FragmentMainBinding.inflate(getLayoutInflater(), container, false);
        return binding.getRoot();
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // Collection is needless here.
        switch (Settings.getTheme().getValue()){
            case FOLLOW_SYSTEM:
                binding.rbFollowSystem.setChecked(true);
                break;

            case DARK:
                binding.rbDark.setChecked(true);
                break;

            case LIGHT:
                binding.rbLight.setChecked(true);
                break;
        }

        binding.rgTheme.setOnCheckedChangeListener(
            (group, checkedId) -> setTheme(checkedId)
        );
    }

    @SuppressLint("NonConstantResourceId")
    private void setTheme(@IdRes int id){
        Theme newTheme;

        switch (id){
            case R.id.rb_followSystem:
                newTheme = Theme.FOLLOW_SYSTEM;
                break;

            case R.id.rb_dark:
                newTheme = Theme.DARK;
                break;

            case R.id.rb_light:
                newTheme = Theme.LIGHT;
                break;

            default:
                throw new IllegalStateException("Unexpected value: " + id);
        }

        Settings.getTheme().setValue(newTheme);
    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: viewkt/app/src/main/java/pers/shawxingkwok/kdatastore/viewkt/app/MainFragment.kt
package pers.shawxingkwok.kdatastore.viewkt.app

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.IdRes
import androidx.fragment.app.Fragment
import pers.shawxingkwok.androidutil.view.collectOnResume
import pers.shawxingkwok.kdatastore.viewkt.settings.Theme
import pers.shawxingkwok.kdatastore.viewkt.settings.Settings
import pers.shawxingkwok.kdatastore.viewkt.app.databinding.FragmentMainBinding

class MainFragment : Fragment() {
    companion object {
        fun newInstance() = MainFragment()
    }

    private var _binding: FragmentMainBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentMainBinding.inflate(layoutInflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Collection is needless here in this case.
        when (Settings.theme.value) {
            Theme.FOLLOW_SYSTEM -> binding.rbFollowSystem
            Theme.DARK -> binding.rbDark
            Theme.LIGHT -> binding.rbLight
        }
        .isChecked = true

        binding.rgTheme.setOnCheckedChangeListener { _, checkedId -> setTheme(checkedId) }
    }

    private fun setTheme(@IdRes id: Int){
        Settings.theme.value =
            when(id){
                R.id.rb_followSystem -> Theme.FOLLOW_SYSTEM
                R.id.rb_dark -> Theme.DARK
                R.id.rb_light -> Theme.LIGHT
                else -> error("")
            }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}